chat-plugins/tickets.js

```diff
@@ -0,0 +1,560 @@
+			}
+
+			if (targetUser) {
+				targetUser.popup(`|modal|${user.name} has banned you from creating help tickets.${(target ? `\n\nReason: ${target}` : ``)}\n\nYour ban will expire in a few days.`);
+			}
+
+			let tbanMessage = `${name} was ticket banned by ${user.name}.${target ? ` (${target})` : ``}`;
+
+			this.addModCommand(tbanMessage, (targetUser ? `(${targetUser.latestIp})` : ''));
+
+			// Notify staff room when a user is ticket banned outside of it.
+			if (room.id !== 'staff' && Rooms('staff')) {
+				Rooms('staff').addLogMessage(user, `<<${room.id}>> ${tbanMessage}`);
+			}
+
+			let affected = [];
+			let punishment = {
+				banned: name,
+				by: user.name,
+				expires: Date.now() + TICKET_BAN_DURATION,
+				reason: target,
+				ip: (targetUser ? targetUser.latestIp : ticket.ip),
+			};
+
+			if (targetUser) {
+				affected.push(targetUser);
+				affected.concat(targetUser.getAltUsers(false, true));
+			} else {
+				let foundKeys = Punishments.search(userid)[0].map(key => key.split(':')[0]);
+				let userids = new Set([userid]);
+				let ips = new Set();
+				for (let key of foundKeys) {
+					if (key.includes('.')) {
+						ips.add(key);
+					} else {
+						userids.add(key);
+					}
+				}
+				affected = Users.findUsers(Array.from(userids), Array.from(ips), {includeTrusted: true, forPunishment: true});
+				affected.unshift(userid);
+			}
+
+			let acAccount = (targetUser && targetUser.autoconfirmed !== userid && targetUser.autoconfirmed);
+			if (affected.length > 1) {
+				this.privateModCommand(`(${name}'s ${acAccount ? ` ac account: ${acAccount}, ` : ""}ticket banned alts: ${affected.slice(1).map(user => user.getLastName()).join(", ")})`);
+			} else if (acAccount) {
+				this.privateModCommand(`(${name}'s ac account: ${acAccount})`);
+			}
+
+			for (let i in affected) {
+				let targetTicket = tickets[(typeof affected[i] !== 'string' ? affected[i].getLastId() : toId(affected[i]))];
+				if (targetTicket) {
+					if (targetTicket.escalated) {
+						if (Rooms('upperstaff')) Rooms('upperstaff').add(`|uhtmlchange|help-ticket-${targetTicket.userid}|`).update();
+					} else {
+						if (Rooms('staff')) Rooms('staff').add(`|uhtmlchange|help-ticket-${targetTicket.userid}|`).update();
+					}
+				}
+				targetTicket = punishment;
+				targetTicket.userid = (typeof affected[i] !== 'string' ? affected[i].getLastId() : toId(affected[i]));
+				targetTicket.name = (typeof affected[i] !== 'string' ? affected[i].getLastName() : `[${toId(affected[i])}]`);
+				if (Rooms('help-' + ticket.userid)) Rooms('help-' + targetTicket.userid).destroy();
+				tickets[(typeof affected[i] !== 'string' ? affected[i].getLastId() : toId(affected[i]))] = targetTicket;
+			}
+			writeTickets();
+
+			this.globalModlog(`TICKETBAN`, targetUser || userid, ` by ${user.name}${target}`);
+			return true;
+		},
+		banhelp: ['/ticket ban [user], (reason) - Bans a user from creating tickets for 2 days. Requires %, @, &, ~'],
```

Comment:
This applies for all help commands, but commas aren't used inthe "Requires !@&#$*(%*&$*#(O($I*U%##%#*&$#U" stuff